### ISSUE TITLE
The plugin is failing when a collection lacks a title

### ISSUE THREAD

**Issue by drnextgis** [stac-utils/qgis-stac-plugin#241]
![image](https://github.com/stac-utils/qgis-stac-plugin/assets/866124/49694c02-6a4d-48bd-a09d-cfc5c22e5746)

**Comment by Samweli** [comment]
@drnextgis thanks for opening this, at the moment when a collection lacks a `title` the plugin defaults to use the collection's `id` which is a mandatory attribute. From the error above it seems the collections also lacks the `id` property.

**Comment by drnextgis** [comment]
That is not right, the collection has `id`. It fails [here](https://github.com/stac-utils/qgis-stac-plugin/blob/2ef210d0ad31c616abf5690a74def057a55691f6/src/qgis_stac/gui/qgis_stac_widget.py#L688) because few lines above `get_selected_collections` method is called with `title=True` argument and as you can see [here](https://github.com/stac-utils/qgis-stac-plugin/blob/2ef210d0ad31c616abf5690a74def057a55691f6/src/qgis_stac/gui/qgis_stac_widget.py#L984) in this case `id` property is just ignored.

**Comment by Samweli** [comment]
> That is not right, the collection has `id`. It fails [here](https://github.com/stac-utils/qgis-stac-plugin/blob/2ef210d0ad31c616abf5690a74def057a55691f6/src/qgis_stac/gui/qgis_stac_widget.py#L688) because few lines above `get_selected_collections` method is called with `title=True` argument and as you can see [here](https://github.com/stac-utils/qgis-stac-plugin/blob/2ef210d0ad31c616abf5690a74def057a55691f6/src/qgis_stac/gui/qgis_stac_widget.py#L984) in this case `id` property is just ignored.

Oh, you are right @drnextgis!
